pkg/restapi/vc/operation/operations.go

```diff
@@ -219,11 +236,83 @@ func New(config *Config) (*Operation, error) {
 	return svc, nil
 }
 
-func createMACCrypto() (ariescrypto.Crypto, *keyset.Handle, string, error) {
-	// TODO: Store HMAC-SHA256 key in persistent store.
-	// If the edge-service instance is shut down, then any previously stored VCs will become unretrievable.
-	// https://github.com/trustbloc/edge-service/issues/237
-	kh, err := keyset.NewHandle(mac.HMACSHA256Tag256KeyTemplate())
+func prepareJWECrypto(config *Config) (string, *authcrypt.Packer, error) {
+	// ToDo: Switch to localKMS. https://github.com/trustbloc/edge-service/issues/309
+	kmsProv := legacyKMSProvider{kms: config.KMS}
+
+	packer := authcrypt.New(kmsProv)
+
+	signingKeyStore, err := prepareSigningKeyStore(config.StoreProvider)
+	if err != nil {
+		return "", nil, err
+	}
+
+	var signingKey string
+
+	signingKeyBytes, err := signingKeyStore.Get(signingKeyDBKeyName)
+	if err != nil {
+		if err == storage.ErrValueNotFound {
+			var createKeySetErr error
+
+			_, signingKey, createKeySetErr = config.KMS.CreateKeySet()
+			if createKeySetErr != nil {
+				return "", nil, createKeySetErr
+			}
+
+			err = signingKeyStore.Put(signingKeyDBKeyName, []byte(signingKey))
+			if err != nil {
+				return "", nil, err
+			}
+		} else {
+			return "", nil, err
+		}
+	} else {
+		signingKey = string(signingKeyBytes)
+	}
+
+	return signingKey, packer, nil
+}
+
+func prepareSigningKeyStore(storeProvider storage.Provider) (storage.Store, error) {
+	err := storeProvider.CreateStore(signingKeyStoreName)
+	if err != nil {
+		if err != storage.ErrDuplicateStore {
+			return nil, err
+		}
+	}
+
+	return storeProvider.OpenStore(signingKeyStoreName)
+}
+
+func prepareCredentialStore(config *Config) (storage.Store, error) {
+	err := config.StoreProvider.CreateStore(credentialStoreName)
+	if err != nil {
+		if err != storage.ErrDuplicateStore {
+			return nil, err
+		}
+	}
+
+	return config.StoreProvider.OpenStore(credentialStoreName)
+}
+
+func prepareMACCrypto(storeProvider storage.Provider,
+	kmsSecretsStoreProvider ariesstorage.Provider) (ariescrypto.Crypto, *keyset.Handle, string, error) {
+	masterKeyReader, err := prepareMasterKeyReader(kmsSecretsStoreProvider)
+	if err != nil {
+		return nil, nil, "", err
+	}
+
+	secretLockService, err := local.NewService(masterKeyReader, nil)
+	if err != nil {
+		return nil, nil, "", err
+	}
+
+	kmsProv := kmsProvider{
+		storageProvider:   kmsSecretsStoreProvider,
+		secretLockService: secretLockService,
+	}
+
+	localKMS, err := localkms.New(masterKeyURI, kmsProv)
```

Comment:
we need localkms for creating did keys

Reply:
first step we need to add localkms for edge-service

Reply:
@Baha-sk and me we work on that

Reply:
OK!

Reply:
@DRK3 create it in start.go

Reply:
done